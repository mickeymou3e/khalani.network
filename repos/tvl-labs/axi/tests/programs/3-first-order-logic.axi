../../tutorial/3-first-order-logic.axi