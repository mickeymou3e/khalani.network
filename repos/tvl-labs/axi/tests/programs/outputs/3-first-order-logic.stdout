Result: Nothing
Result: Nothing
Result: Nothing
Result: Nothing
Result: Nothing
Result: Proposition(exists N. and(Q[@0:N], R[@0:N, @0:N]))
Result: Proved(if(forall N. P[@0:N], forall N. P[@0:N]))
Result: Proved(if(forall N. P[@0:N], forall N. P[@0:N]))
Result: Proved(if(forall N. P[@0:N], forall N. P[@0:N]))
Result: Nothing
Result: Nothing
Result: Nothing
Result: Proved(if(forall N. not(R[@0:N, @0:N]), if(forall N. forall N. forall N. if(and(R[@2:N, @1:N], R[@1:N, @0:N]), R[@2:N, @0:N]), forall N. forall N. if(R[@1:N, @0:N], not(R[@0:N, @1:N])))))
Result: Proved(if(forall N. not(R[@0:N, @0:N]), if(forall N. forall N. forall N. if(and(R[@2:N, @1:N], R[@1:N, @0:N]), R[@2:N, @0:N]), forall N. forall N. if(R[@1:N, @0:N], not(R[@0:N, @1:N])))))
Result: Proved(if(exists N. P[@0:N], exists N. P[@0:N]))
Result: Proved(if(exists N. P[@0:N], exists N. P[@0:N]))
Result: Nothing
Result: Proposition(forall N. and(P[@0:N], Q[@0:N]))
Result: Nothing
Result: Nothing
Result: Proposition(forall N. P[f[f[@0:N]):N]):N])
